src_cpp/opennnBridge/nifppNerltensorEigen.h

```diff
@@ -0,0 +1,127 @@
+#pragma once 
+
+#include "definitionsNN.h"
+#include "nifppEigenExtensions.h"
+
+namespace nifpp
+{
+    template<typename Type>
+    int get_binary(ErlNifEnv *env, ERL_NIF_TERM bin_term, std::vector<Type> &vec);
+
+    int atom_str_to_enum(str_atom in_str);
+    int get_actual_dim(std::vector<int> &dims);
+
+    template<typename BasicType> void get_tensor_data(ErlNifEnv *env , nifpp::TERM binary_term, std::vector<BasicType> &data, std::vector<int> &dims);
+  
+    template<typename Type>
+    int get_binary(ErlNifEnv *env, ERL_NIF_TERM bin_term, std::vector<Type> &vec)
+    {
+        ErlNifBinary bin;
+        int ret = enif_inspect_binary(env, bin_term, &bin);
+        if(!ret)
+        {
+            // a binary either, so fail.
+            return 0;
+        }
+        vec = std::vector<Type>(bin.data, bin.data + bin.size / sizeof(Type));
+        return ret;
+    }
+
+   
+
+    int atom_str_to_enum(str_atom in_str)
+    {
+        if (in_str == "float")
+        {
+            return ATOM_FLOAT;
+        }
+        else if (in_str == "double")
+        {
+            return ATOM_DOUBLE;
+        }
+        else if (in_str == "int32")
+        {
+            return ATOM_INT32;
+        }
+        else if (in_str == "int16")
+        {
+            return ATOM_INT16;
+        }
+        throw("Type is unsupported, update atom_str_to_enum and enum in nifppEigenExtensions.h");
+    }
+
+
+    template<typename BasicType> void get_tensor_data(ErlNifEnv *env , ERL_NIF_TERM bin_term, std::vector<BasicType> &data, std::vector<BasicType> &dims_orig, std::vector<int> &dims)
```

Comment:
Should change to 
get_tensor(ErlNifEnv *env , ERL_NIF_TERM bin_term, TensorTypePtr &tensor_ptr)
Rowmajor binary map to rowmajor tensor swap to colmajor tensor 

make_tensor(ErlNifEnv *env , nifpp::TERM &out_bin_term, TensorTypePtr &tensor_ptr)
convert colmajor --> rowmajor --> binary